torch/_dynamo/allowed_functions.py

```diff
@@ -119,6 +119,37 @@ def _disallowed_function_ids():
         if isinstance(obj, type(torch.FloatStorage))
     ]
     remove += storage
+    c10d_collectives = [
```

Comment:
@mrshenli do you think there is a better way to get a list of c10d collectives?

should we do a torch.distributed.* style ban instead?